src/main/java/teammates/storage/api/FeedbackSessionsDb.java

```diff
@@ -34,47 +32,24 @@
 
     public static final String ERROR_UPDATE_NON_EXISTENT = "Trying to update non-existent Feedback Session : ";
 
+    /**
+     * Gets a list of feedback sessions that have start/end time laid between {@code rangeStart} and {@code rangeEnd}.
+     */
     public List<FeedbackSessionAttributes> getAllOpenFeedbackSessions(Instant rangeStart, Instant rangeEnd) {
-        List<FeedbackSessionAttributes> list = new LinkedList<>();
-
-        // To retrieve legacy data where local dates are stored instead of UTC
-        // TODO: remove after all legacy data has been converted
-        Instant start = rangeStart.minus(Duration.ofHours(25));
-        Instant end = rangeEnd.plus(Duration.ofHours(25));
-
         List<FeedbackSession> endEntities = load()
-                .filter("endTime >", TimeHelper.convertInstantToDate(start))
-                .filter("endTime <=", TimeHelper.convertInstantToDate(end))
+                .filter("endTime >", TimeHelper.convertInstantToDate(rangeStart))
+                .filter("endTime <=", TimeHelper.convertInstantToDate(rangeEnd))
                 .list();
 
         List<FeedbackSession> startEntities = load()
-                .filter("startTime >=", TimeHelper.convertInstantToDate(start))
-                .filter("startTime <", TimeHelper.convertInstantToDate(end))
+                .filter("startTime >=", TimeHelper.convertInstantToDate(rangeStart))
+                .filter("startTime <", TimeHelper.convertInstantToDate(rangeEnd))
                 .list();
 
-        List<FeedbackSession> endTimeEntities = new ArrayList<>(endEntities);
-        List<FeedbackSession> startTimeEntities = new ArrayList<>(startEntities);
-
-        endTimeEntities.removeAll(startTimeEntities);
-        startTimeEntities.removeAll(endTimeEntities);
-        endTimeEntities.addAll(startTimeEntities);
-
-        // TODO: remove after all legacy data has been converted
-        for (FeedbackSession feedbackSession : endTimeEntities) {
-            FeedbackSessionAttributes fs = makeAttributes(feedbackSession);
-            Instant fsStart = fs.getStartTime();
-            Instant fsEnd = fs.getEndTime();
-
-            boolean isStartTimeWithinRange = (fsStart.isAfter(rangeStart) || fsStart.equals(rangeStart))
-                    && fsStart.isBefore(rangeEnd);
-            boolean isEndTimeWithinRange = fsEnd.isAfter(rangeStart) && (fsEnd.isBefore(rangeEnd) || fsEnd.equals(rangeEnd));
-
-            if (isStartTimeWithinRange || isEndTimeWithinRange) {
-                list.add(fs);
-            }
-        }
+        endEntities.removeAll(startEntities);
+        endEntities.addAll(startEntities);
```

Comment:
Is this trying to de-duplicate? If so you may want to consider using `Set` instead, although this works.

Reply:
A bit overkill to wrap the returned list to set and set to list again just to remove duplications. Will add comment here instead.